### ISSUE TITLE
Roslyn.VisualStudio.NewIntegrationTests.CSharp.CSharpSourceGenerators.InvokeNavigateToForGeneratedFile is Flaky

### ISSUE THREAD

**Issue by jmarolf** [dotnet/roslyn#60477]
## Build Information
Build: https://dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_build/results?buildId=1689247
Build error leg: VS_Integration_Debug_64 / Build and Test

## Error Message

*Exception Message*
```
Assert.Equal() Failure
          ↓ (pos 0)
Expected: HelloWorld.cs [generated]
Actual:   Program.cs ⬤
          ↑ (pos 0)
```


<!-- runfo report start -->
Runfo Tracking Issue: [CSharpSourceGenerators.InvokeNavigateToForGeneratedFile is Flaky](https://runfo.azurewebsites.net/tracking/issue/375)
|Build|Definition|Kind|Run Name|
|---|---|---|---|



Build Result Summary
|Day Hit Count|Week Hit Count|Month Hit Count|
|---|---|---|
|0|0|0|

<!-- runfo report end -->

**Comment by jinujoseph** [comment]
cc @sharwell for info

**Comment by jasonmalinowski** [comment]
@JoeRobich Any chance your other fixes for Navigate To might have addressed this?

**Comment by JoeRobich** [comment]
I did update this test as part of the NavigateTo fixes. Opened a PR to see.

**Comment by jasonmalinowski** [comment]
We merged #64626 so we'll see if this is better now.
